packages/did-resolver/docs/DID_Method_Spec.md

```diff
@@ -0,0 +1,769 @@
+DID can be derived given a BNS name, as well as how these DIDs can be updated,
+revoked, and resolved.
+
+# Status of This Document
+
+This document is not a W3C Standard nor is it on the W3C Standards Track. This
+is a draft document and may be updated, replaced or obsoleted by other
+documents at any time. It is inappropriate to cite this document as other than
+work in progress.
+
+Comments regarding this document are welcome. Please file issues directly on
+[Github](https://github.com/blockstack/stacks.js/).
+
+# 1. System Overview
+
+The Blockstack DID method is specified as part of its [decentralized naming
+system](https://docs.stacks.co/build-apps/references/bns) (*BNS*).  Each
+Blockstack name has one corresponding DIDs, and each Blockstack DID corresponds
+to exactly one name -- even if the name was revoked by its owner or expired.
+
+Blockstack is unique among decentralized identity systems in that it is *not*
+anchored to a specific blockchain or DLT implementation.  The system is
+designed from the ground up to be portable, and has already been live-migrated
+from the Namecoin blockchain to the Bitcoin blockchain.  The operational ethos
+of Blockstack is to leverage the must secure blockchain at all times -- that
+is, the one that is considered hardest to attack.
+
+The Blockstack naming system and the derived DIDs transcend the underlying
+blockchain, and will continue to resolve to DID documents even if the system
+migrates to a new blockchain in the future.
+
+## 1.1 DID Lifecycle
+
+Understanding how Blockstack DIDs operate requires understanding how Blockstack
+names operate. Fundamentally, a Blockstack DID is defined as a pointer to a
+*BNS name registered by an address*. The lifecycle of a Blockstack DID is
+therefore tied to the lifecycle of it's underlying BNS name.
+
+The Blockstack naming system differentiates between two types of names (and by
+extension, two types of resulting DIDs), *on-chain* and *off-chain* (elaborated
+on in the following subsections). Both on-chain and off-chain names (as well as
+the resulting DIDs) can be resolved to a set of singing keys, as well as
+additional metadata, using the BNS smart contract, and the Gaia storage
+network, as outlined in section [3.2](#32-resolving-a-stacks-did). 
+
+## 1.2 On-chain DIDs
+
+*On-chain DIDs* are based on [BNS
+names](https://docs.stacks.co/build-apps/references/bns#organization-of-bns),
+the records for which are stored directly on the Stacks blockchain, and managed
+/ updated via interactions with the BNS smart contract. BNS names (and by
+extension on-chain DIDs) can be registered / updated and revoked by calling the
+appropriate functions on the deployed contract. Sections
+[3.1.1](#311-on-chain-names), [3.3.1](#331-unpdating-an-on-chain-stacks-did)
+and [3.4](#34-deactivating-a-stacks-did) outline the respective interactions in
+more detail.
+
+A resolvable Stacks DID can be derived for any on-chain name by concatenating
```

Comment:
What about names owned by a contract? Should these names be excluded? Or should the documentation and lib updated to for names owned by contract principals?